sql/src/main/java/io/crate/expression/operator/LikeOperator.java

```diff
@@ -60,33 +67,48 @@ public FunctionInfo info() {
             if (value == null) {
                 return this;
             }
-            return new CompiledLike(info, (String) value);
+            boolean ignoreCase = ((Literal<Boolean>) arguments.get(2)).value();
+            return new CompiledLike(info, (String) value, ignoreCase);
         }
         return super.compile(arguments);
     }
 
     @Override
     public Boolean evaluate(TransactionContext txnCtx, Input<String>... args) {
         assert args != null : "args must not be null";
-        assert args.length == 2 : "number of args must be 2";
+        assert args.length == 3 : "number of args must be 3";
 
         String expression = args[0].value();
         String pattern = args[1].value();
         if (expression == null || pattern == null) {
             return null;
         }
+        boolean ignoreCase = (Boolean) ((Object) args[2].value());
+        return matches(expression, pattern, ignoreCase);
+    }
+
+    protected static Pattern makePattern(String pattern, boolean ignoreCase) {
+        int flags = Pattern.DOTALL;
+        if (ignoreCase) {
+            flags |= Pattern.CASE_INSENSITIVE;
+        }
+        return Pattern.compile(patternToRegex(pattern, DEFAULT_ESCAPE, true), flags);
+    }
 
-        return matches(expression, pattern);
+    protected static boolean matches(String expression, Pattern pattern) {
+        return pattern.matcher(expression).matches();
```

Comment:
Maybe inline this? These shallow methods might increase complexity due to indirection instead of reducing it. They're also not adding any new semantic meaning.